pkg/snapshot/restorer/restorer.go

```diff
@@ -713,3 +731,53 @@ func verifySnapshotRevision(clientKV client.KVCloser, snap *brtypes.Snapshot) er
 	}
 	return nil
 }
+
+func (r *Restorer) checkRevisionForAll(ro brtypes.RestoreOptions, cfg *brtypes.EtcdConnectionConfig) bool {
+	for _, v := range ro.ClusterURLs {
+		clientFactory := etcdutil.NewClientFactory(ro.NewClientFactory, *cfg)
+		clientKV, err := clientFactory.NewKV()
+		if err != nil {
+			r.logger.Errorf("Could not creat ETCD client for endpoint %s: %v", v[0].String(), err)
+			return false
+		}
+		defer clientKV.Close()
+
+		snapList := ro.DeltaSnapList
+		lastDeltaSnap := snapList[len(snapList)-1]
+		if err := verifySnapshotRevisionForAll(clientKV, lastDeltaSnap); err != nil {
+			r.logger.Errorf("All delta snapshots are not yet applied for %v because : %v", v[0].String(), err)
+			return false
+		}
+	}
+	return true
+}
+
+func verifySnapshotRevisionForAll(clientKV client.KVCloser, snap *brtypes.Snapshot) error {
+	ctx := context.TODO()
+	getResponse, err := clientKV.Get(ctx, "foo")
+	if err != nil {
+		return fmt.Errorf("failed to connect to etcd KV client: %v", err)
+	}
+	etcdRevision := getResponse.Header.GetRevision()
+	if snap.LastRevision < etcdRevision {
+		return fmt.Errorf("this etcd instance has not yet applied all the delta snapshots, expected revision equal or more than %d but applied %d ", snap.LastRevision, etcdRevision)
+	}
+	return nil
+}
+
+func (r *Restorer) isLeader(etcdConnectionConfig *brtypes.EtcdConnectionConfig) bool {
+	factory := etcdutil.NewFactory(*etcdConnectionConfig)
+	clientMaintenance, err := factory.NewMaintenance()
+	if err != nil {
+		r.logger.Errorf("Could not create etcd client while determining if the embedded etcd is leader: %v", err)
+		return false
+	}
+
+	response, err := clientMaintenance.Status(context.Background(), defaultEtcdConnectionEndpoint)
```

Comment:
```suggestion
	response, err := clientMaintenance.Status(context.Background(), etcdConnectionConfig.Endpoints[0])
```
Uses the default endpoint even if one is explicitly set

Reply:
Done